packaging_automation/tests/test_common_tool_methods.py

```diff
@@ -1,201 +1,192 @@
-            prepend_line_in_file(test_file, "^Test line 5", "Test line 4.5")
-
-            with open(test_file, "r") as reader:
-                lines = reader.readlines()
-                self.assertEqual(11, len(lines))
-                self.assertEqual(lines[0], "Test line 0.5\n")
-                self.assertEqual(lines[1], "Test line 1\n")
-                self.assertEqual(lines[2], "Test line 1.5\n")
-                self.assertEqual(lines[3], "Test line 2\n")
-        finally:
-            os.remove(test_file)
-
-
-if __name__ == '__main__':
-    unittest.main()
+def test_find_nth_occurrence_position():
+    assert find_nth_occurrence_position("foofoo foofoo", "foofoo", 2) == 7
+
+
+def test_find_nth_matching_line_number_by_regex():
+    assert find_nth_matching_line_and_line_number("citusx\n citusx\ncitusx", "^citusx$", 2)[0] == 2
+    assert find_nth_matching_line_and_line_number("citusx\n citusx\ncitusx", "^citusy$", 2)[0] == -1
+
+
+def test_is_major_release():
+    assert is_major_release("10.0.0")
+    assert not is_major_release("10.0.1")
+
+
+def test_get_project_version_from_tag_name():
+    tag_name = "v10.0.3"
+    assert "10.0.3" == get_project_version_from_tag_name(tag_name)
+
+
+def test_str_array_to_str():
+    assert "1\n2\n3\n4\n" == str_array_to_str(["1", "2", "3", "4"])
+
+
+def test_run():
+    result = run("echo 'Run' method is performing fine ")
+    assert 0 == result.returncode
+
+
+def test_remove_paranthesis_from_string():
+    assert "out of paranthesis " == remove_text_with_parenthesis("out of paranthesis (inside paranthesis)")
+
+
+def test_get_version_details():
+    assert {"major": "10", "minor": "0", "patch": "1"} == get_version_details("10.0.1")
+
+
+def test_replace_line_in_file():
+    replace_str = "Summary:     Replace Test"
+    copy_file_path = f"{TEST_BASE_PATH}/files/citus_copy.spec"
+    copyfile(f"{TEST_BASE_PATH}/files/citus.spec", copy_file_path)
+    replace_line_in_file(copy_file_path, r"^Summary:	*", replace_str)
+    try:
+        with open(copy_file_path, "r") as reader:
+            content = reader.read()
+            lines = content.splitlines()
+            assert lines[5] == replace_str
+    finally:
+        os.remove(copy_file_path)
+
+def test_get_upcoming_minor_version():
+    assert get_upcoming_minor_version("10.1.0") == "10.2"
+
+def test_get_last_commit_message():
+    current_branch_name = get_current_branch(os.getcwd())
+    test_branch_name = f"test{uuid.uuid4()}"
+    run(f"git checkout -b {test_branch_name}")
+    try:
+        with open(test_branch_name,"w") as writer:
+            writer.write("Test content")
+        run(f"git add .")
+        commit_message = f"Test message for {test_branch_name}"
+        run(f"git commit -m '{commit_message}'")
+        assert get_last_commit_message(os.getcwd()) == f"{commit_message}\n"
+    finally:
+        run(f"git checkout {current_branch_name}")
+        run(f"git branch -D {test_branch_name}")
+
+
+
+
```

Comment:
minor: I think we can remove a few of those new lines, by following what we are normally doing between other functions